Sources/Core/View/UIKit/SnackbarUIView.swift

```diff
@@ -0,0 +1,336 @@
+//
+//  SnackbarUIView.swift
+//  SparkSnackbar
+//
+//  Created by louis.borlee on 04/09/2024.
+//  Copyright © 2024 Adevinta. All rights reserved.
+//
+
+import UIKit
+import Combine
+import SparkTheming
+import SparkButton
+@_spi(SI_SPI) import SparkCommon
+
+/// The UIKit version of the snacbkar.
+public final class SnackbarUIView: UIView {
+
+    /// The spark theme of the snackbar.
+    public var theme: Theme {
+        get {
+            return self.viewModel.theme
+        }
+        set {
+            self.viewModel.theme = newValue
+        }
+    }
+
+    /// The intent of the snackbar.
+    public var intent: SnackbarIntent {
+        get {
+            return self.viewModel.intent
+        }
+        set {
+            self.viewModel.intent = newValue
+        }
+    }
+
+    /// The type of the snackbar.
+    public var type: SnackbarType = .horizontal {
+        didSet {
+            guard self.type != oldValue else { return }
+            self.setNeedsLayout()
+        }
+    }
+
+    /// The variant of the snackbar.
+    public var variant: SnackbarVariant {
+        get {
+            return self.viewModel.variant
+        } set {
+            self.viewModel.variant = newValue
+        }
+    }
+
+    public private(set) var imageView: UIImageView?
+    public let label = UILabel()
+    public private(set) var button: ButtonUIView?
+
+    private var buttonContainer: UIView?
+    private var imageContainer: UIView?
+    private let labelContainer = UIView()
+
+    private let horizontalStackView = UIStackView()
+    private let verticalStackView = UIStackView()
+
+    private let viewModel: SnackbarViewModel
+    private var cancellables = Set<AnyCancellable>()
+
+    @ScaledUIMetric private var scaleFactor: CGFloat = 1.0
```

Comment:
I think it's better to use a @ScaledUIMetric for each properties and not a scaleFactor.
If you test for the cornerWidth, the scaled value is not equals to the cornerWidth * scaleFactor